source/docs/articles/sites/code/redirect-incoming-requests.md

```diff
@@ -30,21 +30,39 @@ Automatic resolution of domains is not supported. For each domain that you want
 
 ## Redirect to a Common Domain
 
-While it’s good for visitors and DNS to resolve both www and the domain itself, it's best practice to choose one or the other and redirect from www to non-www (or vice versa, your call). This optimizes SEO by avoiding duplicate content and prevents session strangeness, where a user can be signed on one domain but logged out of other domains at the same time.  Choose one of the blocks of code below:
-```
-    // Require WWW.
-    if (isset($_SERVER['PANTHEON_ENVIRONMENT']) &&
-      $_SERVER['PANTHEON_ENVIRONMENT'] === 'live') {
-      if ($_SERVER['HTTP_HOST'] == 'yoursite.com' ||
-          $_SERVER['HTTP_HOST'] == 'live-yoursite.pantheon.io') {
-        header('HTTP/1.0 301 Moved Permanently');
-        header('Location: http://www.yoursite.com'. $_SERVER['REQUEST_URI']);
-        exit();
-      }
-    }
+While it’s good for visitors and DNS to resolve both www and the domain itself, it's best practice to choose one or the other and redirect from www to non-www (or vice versa, your call). This optimizes SEO by avoiding duplicate content and prevents session strangeness, where a user can be signed on one domain but logged out of other domains at the same time.
+
+Pantheon's www-redirection service automatically redirects requests to the www subdomain as long as [DNS has been configured](/docs/articles/sites/domains/#step-3-configure-your-dns) appropriately. However, this service does not apply to sites with HTTPS enabled and the following block should be used to standardize on www:
+
 ```
+// Require WWW.
+if (isset($_SERVER['PANTHEON_ENVIRONMENT']) &&
+  $_SERVER['PANTHEON_ENVIRONMENT'] === 'live') {
+  if ($_SERVER['HTTP_HOST'] == 'yoursite.com' ||
+      $_SERVER['HTTP_HOST'] == 'live-yoursite.pantheon.io') {
+    header('HTTP/1.0 301 Moved Permanently');
+    header('Location: http://www.yoursite.com'. $_SERVER['REQUEST_URI']);
+    exit();
+  }
+}
 ```
-    // Remove WWW.
+
+
+
+If you prefer to use the bare domain, use the following code block and run your DNS settings through a service that supports CNAME flattening, such as CloudFlare. For details, see [Cloudflare CDN and DNS on Pantheon](/docs/articles/sites/cloudflare-cdn-and-dns).
+
+<div class="alert alert-info" role="alert">
+<h4>Note</h4>
+If you are running the site on a Pro plan or above with an SSL certificate, use the snippet below without configuring CloudFlare. </div>
+
+To direct all traffic to the bare domain using CloudFlare:
```

Comment:
not "using CloudFlare." -> "you will need to use a domain name host that offers CNAME flattening. See (our doc) section (in `/sites/domains`) and.. add the following..."